app/controllers/posts_controller.rb

```diff
@@ -27,9 +27,26 @@ def update
       render 'edit'
     end
   end
-
+# # search
+#   def search
+#     if query.present?
+#       Post.index
+#     else
+#       scoped
+#     end
+#   end
+
+#   def search
+#     Post.search params[:query]
+#   end
+  
   def index
-    @posts = Post.published.order('views_count DESC , created_at DESC')
+    @posts = Post.all
+    if params[:query].present?
+      @posts = Post.search(params[:query]).order('views_count DESC , created_at DESC')
+    else
+      @posts = Post.published.order('views_count DESC , created_at DESC')
```

Comment:
Agreed with @MacDada . Also do we need to do collect all the posts first before getting to if/else block?